### PR TITLE
[Hotfix] Changed spend amount allocation

### PR DESCRIPTION
**Pull request by noudcorten**
# What has been changed?

* Changed spend amount allocation

# Security Measures

* Throws warning if amount of whitelisted pairs is bigger than the amount of max open trades

### CODE REVIEW

File: modules/stats/tradingmodule.py [hunk 1]
```diff
@@ -102,7 +105,7 @@ def open_trade(self, ohlcv: dict, data_dict: dict) -> None:
             return
 
         # Define spend amount based on realised profit
-        spend_amount = (1. / self.amount_of_pairs) * self.realised_profit
+        spend_amount = (1. / self.max_open_trades) * self.realised_profit
```
Comment (geordle) [inline modules/stats/tradingmodule.py:108]:
```suggestion
        spend_amount = (1. / min(self.max_open_trades, self.amount_of_pairs)) * self.realised_profit
```

Comment (geordle) [reply modules/stats/tradingmodule.py:108]: @DemianVoorhagen6 I overheard you mentioning this warning. We could use the `min` of  `max_open_trades, amount_of_pairs` instead.

**Review by geordle — APPROVED** [review]

File: modules/stats/tradingmodule.py [hunk 2]
```diff
@@ -24,6 +24,9 @@ def __init__(self, config: TradingModuleConfig):
 
         self.max_open_trades = int(self.config.max_open_trades)
         self.amount_of_pairs = len(self.config.pairs)
+        if self.amount_of_pairs < self.max_open_trades:
+            print_warning("Not all trading funds are used; amount of whitelisted pairs is smaller than max open trades.")
```
Comment (geordle) [inline modules/stats/tradingmodule.py:28]:
```suggestion
        if self.amount_of_pairs < self.max_open_trades:
            print_warning("max_open_trades exceeds amount of pairs in whitelist. Open trades capped to amount of pairs.")
```

Comment (DemianVoorhagen6) [reply modules/stats/tradingmodule.py:28]: I guess that this makes a bit more sense?
print_warning("max_open_trades exceeds amount of pairs in whitelist. max_open_trades is set to amount of pairs in whitelist.")

Comment (geordle) [reply modules/stats/tradingmodule.py:28]:
```suggestion
        if self.amount_of_pairs < self.max_open_trades:
            print_warning("max_open_trades exceeds amount of pairs in whitelist. max_open_trades will be limited to the amount of pairs in whitelist.")
```

**Review by geordle — DISMISSED** [review]

File: modules/stats/tradingmodule.py (hunk 2 above)
Comment (geordle) [inline modules/stats/tradingmodule.py]:
```suggestion
            print_warning("max_open_trades exceeds amount of pairs in whitelist. max_open_trades will be limited to the amount of pairs in whitelist.")
```

**Pull request merged**